### ISSUE TITLE
This Extension cannot be published as it has the same App ID and Version as a previously published Extension

### ISSUE THREAD

**Issue by RodrigoPuelma** [microsoft/AL#7721]
Please include the following with each issue:

**1. Describe the bug**
When trying to publish an extension via VS Code to a clean sandbox, I get the following error:

"The request for path [] failed with code UnprocessableEntity. Reason: This Extension cannot be published as it has the same App ID and Version as a previously published Extension."

The sandbox is created in a tenant where the Production environment has current version 23.5.16502.16589 while the sandbox is created with the next version 24.0.16410.17628.

The extension versions are different as we are trying to do some testing for the next version of Business Central. This has never been an issue before.

**2. To Reproduce**
1. Create new sandbox environment with version: AU Business Central 24.0 (Platform 24.0.18532.0 + Application 24.0.16410.17628)
2. Run to publish app.

**3. Expected behavior**
No errors when publishing app via VS Code.

**4. Actual behavior**
Error message:
"The request for path [] failed with code UnprocessableEntity. Reason: This Extension cannot be published as it has the same App ID and Version as a previously published Extension."

**5. Versions:**
AU Business Central 24.0 (Platform 24.0.18532.0 + Application 24.0.16410.17628)

### Final Checklist

Please remember to do the following:

* [X] Search the issue repository to ensure you are reporting a new issue

* [X] Reproduce the issue after disabling all extensions except the AL Language extension

* [X] Simplify your code around the issue to better isolate the problem

**Comment by RodrigoPuelma** [comment]
Update:
After removing other sandbox environments and publishing again, I instead get the error below:

Reason: No published extensions match the specified parameter values. 

When logging in I can see the extension published but not installed in the extension management. If I try to publish yet again, I continue getting the same error above.

**Comment by YvesDesrumaux** [comment]
we encounter exactly the same issue.

**Comment by cghasvi** [comment]
same issue here too

**Comment by ENDO261985** [comment]
Same thing happens to me too, even if in the development sandbox you try to load the app with the load extensions.
If you compile from vs code the app is only published and not installed and if you do the manual installation operation it works. This is a Microsofts bug.

**Comment by EmilDamsbo** [comment]
Hello, can you please provide some context on the state of your sandbox as you are trying to publish from VSCode?

1. Is the currently installed version installed as a global app, a PTE, or a DEV extension?
2. Does this happen only when publishing from VSCode as a DEV extension, and not via the extension management page?
3. What happens if you uninstall the currently installed version, then re-install it, and then publish from VScode?

We recommend you to open a support case in order to get the necessary information about your environment. Please refer customer support to this GitHub issue if you do open the support case.

**Comment by ENDO261985** [comment]
Hi, if I'm publishing from vs code it means I'm developing on a sandbox, not a PRODUCTION.
Even with the loading extensions I have problems, even if I unpublish the app and have a higher version of the app.
Business central SAAS 24 wave 1.

**Comment by cghasvi** [comment]
I did this test to reproduce: 
1. Create new, fresh  Sandbox - version 24.0.16410.17628 lang: NO
2. Create new extension in Visual Code with AL:GO!
3. Try to publish the new extension to new Sandbox with CTR+F5 or F5
4. Error in VB Code . failed with code UnprocessableEntity. Reason: Ingen publiserte utvidelser samsvarer med de angitte parameterverdiene.  (last part in Norwegian for some reason but same error as stated before) 
5. Extension is published to environment, but not installed
6. I can install the published extension within BC "Installed Extension" page
7. Main problem for me is that I cannot debug in version 24 now

**Comment by KarelKoncel** [comment]
We have same issue from friday and hoping. but we can't even install from extension page. I hope that it will be fixed soon, this is really big error.

**Comment by RodrigoPuelma** [comment]
> Hello, can you please provide some context on the state of your sandbox as you are trying to publish from VSCode?
> 
> 1. Is the currently installed version installed as a global app, a PTE, or a DEV extension?
> 2. Does this happen only when publishing from VSCode as a DEV extension, and not via the extension management page?
> 3. What happens if you uninstall the currently installed version, then re-install it, and then publish from VScode?
> 
> We recommend you to open a support case in order to get the necessary information about your environment. Please refer customer support to this GitHub issue if you do open the support case.

Hello,

Thank you for your reply.

1. The extension is installed from VS Code. The version installed in the PROD is a published, previous version (23).
2. Impossible to know as I am pushing to a version that is a later version (24) and not available for production (I believe it is today 22/04/2024).
3.  It never gets installed, so there is nothing to uninstall. 
4. I have raised a question in Yammer. Considering no development cant be done by devs there is quite an urgency in having this issue solved.

**Comment by RhipeDynamics** [comment]
I'm getting the same issues too.

Suggestion:
For deployment use the Extension Management interface 
Just for Debugging, you can add a configuration for Attach in the launch.json, Example below
        {
            "name": "Attach: Microsoft cloud sandbox",
            "type": "al",
            "request": "attach",
            "environmentType": "Sandbox",
            "environmentName": "sandbox",
            "breakOnError": "All",
            "breakOnRecordWrite": "None",
            "enableSqlInformationDebugger": true,
            "enableLongRunningSqlStatements": true,
            "longRunningSqlStatementsThreshold": 500,
            "numberOfSqlStatements": 10,
            "breakOnNext": "WebClient"
        }

**Comment by cvicq** [comment]
The same issue since 18/04 for me.

**Comment by cghasvi** [comment]
Thank you @RhipeDynamics . The "attach" workaround is working and helps with debugging.

**Comment by Prachii202** [comment]
same issue facing since yesterday and after upgrading app id and version it shows 
![image](https://github.com/microsoft/AL/assets/107498845/f8b8fa3d-ca23-4d3f-b0dc-002a2e002c52)

**Comment by nick-webbgh** [comment]
The same issue for me.

**Comment by bricetaullee** [comment]
Exactly the same issue for me.

**Comment by frankjanssenwewo** [comment]
Same issue here, right now different/new error:
![image](https://github.com/microsoft/AL/assets/69365719/9340ffdd-d42b-44cc-8708-30476f6176fb)

**Comment by ashishsharmainecta** [comment]
Same Issue for me, I am publishing the app as PTE and still it error with "**Deployment failed. Errors: App ID : ******* Message : { An app with package ID '*******' is already published for tenant 'msft1a6720t93294803' } - Job Id : *********" . On extension management it is published, but not installed. Then I am able to install it successfully from there.

**Comment by frottke** [comment]
Same issue for us, is there an workaround? That is not acceptable...

**Comment by JorgeCarlos** [comment]
I already tested this problem in 3 different customer tenants running version 24 and all of them has the same problem.
I tested in version 23.5 and there are no issues. 
I recommend that you don't upgrade to version 24 until this critical issues is fixed. 
If you need to do an upgrade execute a backup or copy your 23.5 version to a sandbox before an upgrade.

**Comment by tristancliffe** [comment]
I had a preview environment that worked fine. 
I initially upgraded our Sandbox to V24 and didn’t have any issues updating extensions via VS Code.

It was only AFTER I committed to V24 on Production that the problem started. I could roll back, but it’s a hassle having to set up external data links and synchronise with online shops that I won’t for something this non-serious.

**Comment by RhipeDynamics** [comment]
> Same issue for us, is there an workaround? That is not acceptable...
Workaround:
https://github.com/microsoft/AL/issues/7721#issuecomment-2071421265

**Comment by kalberes** [comment]
@tristancliffe do you have a version number for your preview environment where things have worked?

**Comment by ENDO261985** [comment]
23.5 SAAS

**Comment by tristancliffe** [comment]
Erm...

- Preview environment (Chronus and My Company) was 24.0.16410.16682 which I used to test a few things in the environment and with AL Code.
- Sandbox was updated to 24.0.16410.17628 and I'm pretty sure that I pushed an update from VS Code on Sunday evening that worked. It is still that version and doesn't work now - but if I push an update a few times I can install it manually from Extension Management...

**Comment by DmitryKatson** [comment]
Same issue for my US saas sandbox environment of v24

**Comment by tinfisterNP** [comment]
Same issue here, DK 24 sandbox.

ForceSync not working on 24 sandbox. We have tested, it's still working on 23.5 sandbox

**Comment by developerchristiaens** [comment]
Same issue

**Comment by jmarinLaberit** [comment]
Also getting the same issue here.

**In addition**: cannot Copy a Sandbox environment into production (obviously not yet a working company). Resulting production environment has only 'Cronus' and 'My comany' companies. Not the ones in the original sandbox environment.

**Comment by thegaet** [comment]
Same here

**Comment by POG2I** [comment]
Same

**Comment by EBCPKA** [comment]
Same here

**Comment by CBIZGIT** [comment]
Thanks for placing a notice to acknowledge on AL Home.

Have same issue with VSCode dev, and API /extensionUpload(0)/content

Publish phase succeeds , Install phase gets error...  { An app with package ID 'xxxxxxxxxxxxx' is already published for tenant...

Subsequent manual install from extensions page OK.

Affected Sandbox and Production deployments. BC24 and for me BC23.5 following BC24 failed update. (we are GB)

Look forward to fix - Thanks

**Comment by pratamarendi-apac** [comment]
**Alt Solution  :** for workaround (waiting until stable version) , if on your production still v23, 
i think u need copy from your production to sandbox then you can continue with v23 until v24 stable or another version 
that already stable from Microsoft.

**Comment by frankjanssenwewo** [comment]
Finally this anoying issue made it to Microsofts attention and got "Issue" Status: ;)
https://admin.microsoft.com/AdminPortal/Home#/servicehealth/:/alerts/DN784551

**Comment by frankjanssenwewo** [comment]
It´s alive: 
https://admin.microsoft.com/AdminPortal/Home#/servicehealth/:/alerts/DN784551 
Deployment is on the way, Hopefully completed over the weekend.

**Comment by EmiPellic** [comment]
Same issue here, and the problem is when you publish an app with dependencies.
in that case the workaround doesn't work and also publishing phase is failing:
_[2024-04-27 07:26:34.64] The request for path /v2.0/ADV_DEMO/dev/apps?tenant=xxxxxxxxxxxxxxxxxxxxxxxxxxxxx&SchemaUpdateMode=forcesync&DependencyPublishingOption=default failed with code UnprocessableEntity. Reason: This Extension cannot be published as it has the same App ID and Version as a previously published Extension._

**Comment by cghasvi** [comment]
I am now able to publish and debug again

**Comment by frankjanssenwewo** [comment]
Yeah, seems to work for me, too. I was able to publish and debug from VS Code. Also manual installation of extension via extension management works again. Hope it will last :)

**Comment by BazookaMusic** [comment]
I'm closing the issue as completed since this is an issue with the service itself and not the AL compiler. We will also release an update through the official channels for when the fix is completely deployed

**Comment by jackgentle** [comment]
I am still seeing the errors too.  I've tried with three of my customers' BC tenants.  All the same behavior.

**Comment by RSMNishant** [comment]
any estimated time frame for hotfix to be deployed.

**Comment by tristancliffe** [comment]
It is out in the wild, but takes time to reach all environments - some people had it earlier (see answers above saying all is okay, others like me were a couple of hours later etc. I'd imagine it won't take long to propogate, but generally the guide rule of thumb seems to be 24 hours.

**Comment by EmilyH4** [comment]
Does anyone know the version number(s) for the hotfix?  
Admin Center says my environment is Up to Date, but it has the same publishing problem as last week.

**Comment by cghasvi** [comment]
I do not think version number is changing with the fix. I have 24.0.16410.17628 that had the error and is working now.

**Comment by RSMNishant** [comment]
fix is still not available....

**Comment by RSMNishant** [comment]
> fix is still not available....

failed with code UnprocessableEntity. Reason: Sequence contains more than one element

I am seeing this message now.

**Comment by EmilyH4** [comment]
> > fix is still not available....
> 
> failed with code UnprocessableEntity. Reason: Sequence contains more than one element
> 
> I am seeing this message now.

I cleared alpackages and cache, then downloaded symbols from the updated environment.  After that mine worked.

**Comment by HazardMK** [comment]
Hi @RodrigoPuelma  
Why is the status of this issue Closed? People are still getting errors...

**Comment by RSMNishant** [comment]
> > > fix is still not available....
> > 
> > 
> > failed with code UnprocessableEntity. Reason: Sequence contains more than one element
> > I am seeing this message now.
> 
> I cleared alpackages and cache, then downloaded symbols from the updated environment. After that mine worked.

Thanks but still not working... I think fix has not been  released yet for some region.

**Comment by ENDO261985** [comment]
they have already replied that they are working on it, there is no point in writing anymore.

**Comment by EmilDamsbo** [comment]
@RSMNishant 
> Hi @RodrigoPuelma Why is the status of this issue Closed? People are still getting errors...

Please refer to the previous reply here:
> https://github.com/microsoft/AL/issues/7721#issuecomment-2082288284

It is the policy of this repository to close issues once a fix has been planned for release. People with the updated BC platform version are reporting being unblocked., so please be patient as we safely release the fix in stages.
